File: CS/CS/E911.Module.Win/Editors/WinCustomUserControlViewItem.cs
using System;
using E911.Module.Editors;
using System.ComponentModel;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Editors;

namespace E911.Module.Win.Editors {
    public interface IModelWinCustomUserControlViewItem {
        [Category("Data")]
        string ControlTypeName { get; set; }
    }
    [ViewItem(typeof(IModelCustomUserControlViewItem))]
    public class WinCustomUserControlViewItem : CustomUserControlViewItem {
        protected IModelWinCustomUserControlViewItem model;
        public WinCustomUserControlViewItem(IModelViewItem model, Type objectType)
            : base(model, objectType) {
            this.model = model as IModelWinCustomUserControlViewItem;
            if (this.model == null)
                throw new ArgumentNullException("IModelWinCustomUserControlViewItem must extend IModelCustomUserControlViewItem in the ExtendModelInterfaces method of your Win ModuleBase descendant.");
        }
        protected override object CreateControlCore() {
            //You can access the View and other properties here to additionally initialize your control.
            return DevExpress.Persistent.Base.ReflectionHelper.CreateObject(model.ControlTypeName);
        }
    }
}

File: VB/E911.Module.Win/Editors/WinCustomUserControlViewItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports E911.Module.Editors
Imports System.ComponentModel
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Editors

Namespace E911.Module.Win.Editors
	Public Interface IModelWinCustomUserControlViewItem
		<Category("Data")> _
		Property ControlTypeName() As String
	End Interface
	<ViewItem(GetType(IModelCustomUserControlViewItem))> _
	Public Class WinCustomUserControlViewItem
		Inherits CustomUserControlViewItem
		Protected model As IModelWinCustomUserControlViewItem
		Public Sub New(ByVal model As IModelViewItem, ByVal objectType As Type)
			MyBase.New(model, objectType)
			Me.model = TryCast(model, IModelWinCustomUserControlViewItem)
			If Me.model Is Nothing Then
				Throw New ArgumentNullException("IModelWinCustomUserControlViewItem must extend IModelCustomUserControlViewItem in the ExtendModelInterfaces method of your Win ModuleBase descendant.")
			End If
		End Sub
		Protected Overrides Function CreateControlCore() As Object
			'You can access the View and other properties here to additionally initialize your control.
			Return DevExpress.Persistent.Base.ReflectionHelper.CreateObject(model.ControlTypeName)
		End Function
	End Class
End Namespace